lib/datapackage.py

```diff
@@ -0,0 +1,335 @@
+                        except:
+                            print("Exception occured. Try Again.\n")
+                            val = input(d_opts[i]+": ")
+
+
+            table['schema'] = {}
+            table['schema']["fields"] = []
+            print("Enter columns [format = name, type, (optional) size]:\n")
+            col = input()
+            while col != "":
+                try:
+                    col_list = [c.strip() for c in col.split(",")]
+                    col_list = [v for v in col_list if v!=""]
+
+                    col_obj = {}
+                    col_obj["name"] = col_list[0]
+                    col_obj["type"] = col_list[1]
+
+                    if len(col_list) > 2:
+                        if type(eval(col_list[2])) != int:
+                            raise
+                        col_obj["size"] = col_list[2]
+                    table["schema"]["fields"].append(col_obj)
+
+                except:
+                    print("Exception occured. Check the input format again.\n")
+                    pass
+
+                col = input()
+
+
+            isCT = input("Add crosstab columns? (y,N): ")
+            if isCT.lower() in ["y", "Y"]:
+                ct_column = input("Crosstab column name: ")
+                while ct_column == "":
+                    print("Empty column name. Try again.\n")
+                    ct_column = input("Crosstab column name: ")
+                ct_names = []
+                print("Enter names of crosstab column values (Press return after each name)")
+                name = input()
+                while name != "":
+                    ct_names.append(name)
+                    name = input()
+
+                table['schema']['ct_column'] = ct_column
+                table['schema']['ct_names'] = ct_names
+
+            contents['resources'].append(table)
+
+    file_name = contents['name'] + ".json"
+    with open(os.path.join(HOME_DIR, 'scripts', file_name), 'w') as output_file:
+        json.dump(contents, output_file, sort_keys=True, indent=4,
+            separators=(',', ': '))
+        output_file.write('\n')
+        print("\nScript written to "+file_name)
+        output_file.close()
+
+
+def edit_dict(obj, tabwidth=0):
+    '''
+    Recursive helper function for edit_json() to edit a datapackage.JSON script file.
+    '''
+    for (key, val) in obj.items():
+        print('\n'+"  "*tabwidth+"->"+key+" : \n")
+        if type(val) == list:
+            for v in val:
+                print("  "*tabwidth+str(v)+'\n')
+        elif type(val) == dict:
+            for item in val.items():
+                print("  "*tabwidth+str(item)+'\n')
+        else:
+            print("  "*tabwidth+str(val)+'\n')
+
+        while True:
+            try:
+                if isinstance(val,dict):
+                    print("\n\t'"+key+"' has the following keys:\n"+str(obj[key].keys())+"\n")
```

Comment:
for the spaces in between operators, if there is a reason for that then ignore them 👍